### DISCUSSION TITLE
SQLAlchemy crashing python, when executing query

### DISCUSSION THREAD

**Discussion by s-pratyush** [sqlalchemy/sqlalchemy#10812] [Usage Questions]
### Describe the bug

My python-interpreter is crashing  when executing/commiting any simple query .
```
(se-mono) (base) pratyush@Pratyushs-MacBook-Pro se-mono % python
Python 3.11.4 (main, Jan  1 2024, 13:32:34) [Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from sqlalchemy import create_engine, text
>>> from sqlalchemy.orm import scoped_session, sessionmaker
>>> url = "postgresql://download_service_user:***@localhost:5432/download_service"
>>> engine = create_engine(
...     url,
...     echo=False,
...     pool_size=5,
...     max_overflow=10,
...     pool_timeout=60,
...     pool_recycle=3600,
... )
>>> scoped_db_session = scoped_session(
...     sessionmaker(
...         autocommit=False,
...         autoflush=False,
...         bind=engine,
...     ),
... )
>>> session = scoped_db_session
>>> session.execute(text("SELECT 1;"))
python(71704,0x1dbe35000) malloc: double free for ptr 0x13b808200
python(71704,0x1dbe35000) malloc: *** set a breakpoint in malloc_error_break to debug
zsh: abort      python
```

### SQLAlchemy Version in Use

2.0.24

### DBAPI (i.e. the database driver)

psycopg2

### Database Vendor and Major Version

postgresql 13

### Python Version

3.11 above

### Operating system

mac

### To Reproduce

```python
(se-mono) (base) pratyush@Pratyushs-MacBook-Pro se-mono % python
Python 3.11.4 (main, Jan  1 2024, 13:32:34) [Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from sqlalchemy import create_engine, text
>>> from sqlalchemy.orm import scoped_session, sessionmaker
>>> url = "postgresql://download_service_user:****@localhost:5432/download_service"
>>> engine = create_engine(
...     url,
...     echo=False,
...     pool_size=5,
...     max_overflow=10,
...     pool_timeout=60,
...     pool_recycle=3600,
... )
>>> scoped_db_session = scoped_session(
...     sessionmaker(
...         autocommit=False,
...         autoflush=False,
...         bind=engine,
...     ),
... )
>>> session = scoped_db_session
>>> session.execute(text("SELECT 1;"))
python(71704,0x1dbe35000) malloc: double free for ptr 0x13b808200
python(71704,0x1dbe35000) malloc: *** set a breakpoint in malloc_error_break to debug
zsh: abort      python
```

### Error

python(71704,0x1dbe35000) malloc: double free for ptr 0x13b808200
python(71704,0x1dbe35000) malloc: *** set a breakpoint in malloc_error_break to debug
zsh: abort      python

**Comment by zzzeek** [comment]
hi -

you might possibly have this issue: https://github.com/python-greenlet/greenlet/issues/382   upgrade greenlet to the latest

overall this issue would not be within SQLAlchemy and would involve either the python interpreter in use or the installation of native dependencies like psycopg2 or greenlet.

**Comment by zzzeek** [comment]
try this Python script:


```py

from sqlalchemy import create_engine

e = create_engine("sqlite://", echo=True)

with e.connect() as conn:
    print(conn.exec_driver_sql("select 1").all())
```

next, install [pg8000](https://github.com/tlocke/pg8000), pure python postgresql driver:

```
pip install pg8000
```

then run your original script modifying your URL to use pg8000:

```py
from sqlalchemy import create_engine, text
from sqlalchemy.orm import scoped_session, sessionmaker
url = "postgresql+pg8000://download_service_user:***@localhost:5432/download_service"


engine = create_engine(
     url,
     echo=False,
     pool_size=5,
     max_overflow=10,
     pool_timeout=60,
     pool_recycle=3600,
 )
scoped_db_session = scoped_session(
     sessionmaker(
         autocommit=False,
         autoflush=False,
         bind=engine,
     ),
 )
session = scoped_db_session
result = session.execute(text("SELECT 1;"))
print(result.all())

```

if you get that far, then you have a psycopg2 installation issue and you'd want to check that you have a proper match between your psycopg2 build and your current installed [libpq](https://www.postgresql.org/docs/9.5/libpq.html)

**Reply by s-pratyush** [reply]
@zzzeek I tried after using `pg8000`, got the below error.

`sqlalchemy.exc.InterfaceError: (pg8000.exceptions.InterfaceError) {'S': 'FATAL', 'V': 'FATAL', 'C': '28000', 'M': 'no pg_hba.conf entry for host "223.233.71.235", user "download_service_user", database "download_service", no encryption', 'F': 'auth.c', 'L': '647', 'R': 'ClientAuthentication'}
`

**Reply by zzzeek** [reply]
OK this is postgresql configuration stuff.  

https://dba.stackexchange.com/questions/83984/connect-to-postgresql-server-fatal-no-pg-hba-conf-entry-for-host should get you started.

**Reply by s-pratyush** [reply]
@zzzeek I am using remote azure-pg-server and my IP is already whitelisted.

**Comment by painswor28** [comment]
I just got this same error when using SQLAlchemy with postgres.  If you uninstall `psycopg2` and install `psycopg2-binary` as the underlying postgres dependency it should fix the error.
